app/battle_preparation.py

```diff
@@ -0,0 +1,45 @@
+from __future__ import annotations
+
+
+class KnightPreparation:
+
+    def __init__(self,
+                 name: str,
+                 hp: int,
+                 power: int,
+                 armour: list,
+                 weapon: dict,
+                 potion: dict | None) -> None:
+        self.name = name
+        self.hp = hp
+        self.power = power
+        self.armour = armour
+        self.weapon = weapon
+        self.potion = potion
+        self.protection = 0
+
+    def apply_armour(self) -> None:
+        for ar in self.armour:
+            self.protection += ar["protection"]
+
+    def apply_weapon(self) -> None:
+        self.power += self.weapon["power"]
+
+    def apply_potion(self) -> None:
+        if self.potion is not None:
+            if "power" in self.potion["effect"]:
+                self.power += \
+                    self.potion["effect"]["power"]
+
+            if "protection" in self.potion["effect"]:
+                self.protection += \
+                    self.potion["effect"]["protection"]
+
+            if "hp" in self.potion["effect"]:
+                self.hp += \
+                    self.potion["effect"]["hp"]

```

Comment:
DRY. Use a for loop to simplify you code

Reply:
i guess i need a hint...
so i can write a cycle:

```
for property in self.potion["effect"]:
      self.property += self.potion["effect"][property]
```

but i know there'll be a conflict about "self.property" (because there's no such property, only self.hp, self.power and self.protection)
how can i prevent that in cycle?